container.go

```diff
@@ -0,0 +1,342 @@
+package hcsshim
+
+import (
+	"encoding/json"
+	"syscall"
+
+	"errors"
+	"github.com/Sirupsen/logrus"
+	"time"
+)
+
+type hcsShimContainer struct {
+	handle HcsSystem
+}
+
+// CreateContainer creates a new container with the given configuration but does not start it.
+func CreateContainer(id string, c *ContainerConfig) (Container, error) {
+	title := "HCSShim::CreateContainer"
+	var (
+		handle   HcsSystem
+		identity syscall.Handle
+		resultp  *uint16
+	)
+
+	configurationb, err := json.Marshal(c)
+	if err != nil {
+		return nil, err
+	}
+
+	configuration := string(configurationb)
+
+	if err := hcsCreateComputeSystem(id, configuration, identity, &handle, &resultp); err != nil {
+		err = makeErrorf(err, title, "id=%s configuration=%s", id, configuration)
+		logrus.Error(err)
+		return nil, err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
```

Comment:
Does convertAndFreeCoTaskMemString handle a value of nil already? If not, can we make it do so?